pyiceberg/schema.py

```diff
@@ -324,6 +324,11 @@ def field_ids(self) -> Set[int]:
         """Return the IDs of the current schema."""
         return set(self._name_to_id.values())
 
+    @property
+    def field_names(self) -> Set[str]:
```

Comment:
Preferably I would not expose this field, since there are some issues with it:

Conflicts with structs:
```
schema {
  1: a.b string
  2: a   struct<1: b string>
}
```

Or with lists:

```
schema {
  1: a.element string
  2: a List<3: element string>
}
```

Reply:
Sounds good - I've removed the property, since it's only being used in `_check_pyarrow_schema_compatible`